R/utils.R

```diff
@@ -65,3 +65,23 @@ assert_global_is_null <- function(key, state) {
     stop("global variable <", key, "> in <state> was not NULL")
   }
 }
+
+
+tidy_user_info <- function(user_info) {
+
+  while(purrr::every(user_info, function(x) length(x) == 1)){
```

Comment:
An alternative to 

```r
  while(purrr::every(user_info, function(x) length(x) == 1)){
    user_info <- purrr::map(user_info, function(x) {
      if(length(x) > 1) return(x) else return(paste0(x, collapse = ", "))
    })
  }
```

would be 

```r
user_info <- map(user_info, jsonlite::toJSON)
```

It's shorter, which is always good, but it also has the advantage that the user can easily reverse the operation to get back vectors by using `jsonlite::fromJSON`. What do you think?

Reply:
Sure. The only issue with that is that it wraps unnecessary JSON around scalar values and stores the data as Javascript-like, rather than R-like, when we are in R world:

<img width="422" alt="Screenshot 2024-06-25 at 10 06 57" src="https://github.com/pmcharrison/psychTestR/assets/39628712/f69265f5-ecc5-4dc9-a34c-a313ff806a08">


As a hybrid in the middle, how about:

```r
 user_info <- purrr::map(user_info, function(x) {
    if(length(x) > 1) jsonlite::toJSON(x) else x
  })
  ```
  
  This way, we don't store scalar values in the `session` object with unnecessary JSON formatting, but for vectors with length > 1, we can coerce them back into vectors from their JSON string representations. e.g.,:
  
  
<img width="645" alt="Screenshot 2024-06-25 at 10 04 45" src="https://github.com/pmcharrison/psychTestR/assets/39628712/f8ca2914-8d8d-441a-a525-c27b50f73a02">

Reply:
Hi @sebsilas, thanks for this! Hmm, good point. You might be able to alternatively address this by passing `auto_unbox = TRUE` to `toJSON`? Documentation below:

> automatically unbox() all atomic vectors of length 1. It is usually safer to avoid this and instead use the unbox() function to unbox individual elements. An exception is that objects of class AsIs (i.e. wrapped in I()) are not automatically unboxed. This is a way to mark single values as length-1 arrays.

Potentially this solves some rough edges in the screenshot you posted (e.g. timeZones is not ideal yet). But your proposal works fine if that doesn't work.

Reply:
Hm nice idea, although that also doesn't leave some things looking very nice:

<img width="1161" alt="Screenshot 2024-07-01 at 16 45 31" src="https://github.com/pmcharrison/psychTestR/assets/39628712/801872ef-c4fb-40e3-a085-5b811bc38691">



<img width="1071" alt="Screenshot 2024-07-01 at 16 45 16" src="https://github.com/pmcharrison/psychTestR/assets/39628712/6b95940d-345d-4ef1-9b0e-36944d0747bb">


I agree that timeZones isn't ideally structured JSON in my approach, but at least it really does coerce back an R character vector:

```r
test <- rjson::fromJSON('[["Europe/Berlin"],["Europe/Busingen"]]')
```

I'm going to submit a PR with my approach for now, but let me know if you have any other ideas